### PR TITLE
YUIdoc comments in docpad.coffee up to Template data section

### PR DESCRIPTION
**Pull request by SteveMcArthur**

### CODE REVIEW

File: src/lib/docpad.coffee [hunk 1]
```diff
@@ -135,7 +135,8 @@ class DocPad extends EventEmitterGrouped
 	# ---------------------------------
 	# Modules
 
-	# Base-----------------------------
+	# Base
+	# ---------------------------------
```
Comment (balupton) [inline src/lib/docpad.coffee]: Should be:

```
	# ---------------------------------
	# Base
```

File: src/lib/docpad.coffee [hunk 2]
```diff
@@ -579,42 +772,91 @@ class DocPad extends EventEmitterGrouped
...
 	getUrlPathname: (url) ->
 		return url.replace(/\?.*/,'')
 
-	# Get a file by its route
+
 	# next(err,file)
+	###*
+	# Get a file by its route
+	# @method getFileByRoute
+	# @param {Object} url
```
Comment (balupton) [inline src/lib/docpad.coffee]: String

**Comment by balupton** [comment]
Address the two comments, and will merge.

Also, are there Mixed or Unknown types in YUIDoc?

**Comment by SteveMcArthur** [comment]
Just checked the yui syntax documentation (http://yui.github.io/yuidoc/syntax/). Yes - there is a "mixed" type and also "any".

File: src/lib/docpad.coffee [hunk 3]
```diff
@@ -159,7 +161,8 @@ class DocPad extends EventEmitterGrouped
 	###
 	QueryCollection: QueryCollection
 
-	# Models----------------------------
+	# Models
+	# ---------------------------------
```
Comment (balupton) [inline src/lib/docpad.coffee]: swap here too

File: src/lib/docpad.coffee [hunk 4]
```diff
@@ -172,7 +175,8 @@ class DocPad extends EventEmitterGrouped
 	###
 	DocumentModel: DocumentModel
 
-	# Collections------------------------
+	# Collections
+	# ---------------------------------
```
Comment (balupton) [inline src/lib/docpad.coffee]: swap here too

File: src/lib/docpad.coffee [hunk 5]
```diff
@@ -646,8 +889,15 @@ class DocPad extends EventEmitterGrouped
...
 		@
 
-	# Get a file by its selector
+
 	# TODO: What on earth is a selector?
+	###*
+	# Get a file by its selector
+	# @method getFileBySelector
+	# @param {Object} selector
+	# @param {Object} opts={}
```
Comment (balupton) [inline src/lib/docpad.coffee]: `# @param {Object} [opts={}]`

**Pull request merged**